### PR TITLE
clang-tidy fix

### PR DESCRIPTION
**Pull request by AlCutter**
Just the output of clang-tidy over all the `.cc` and `.h` files.

### CODE REVIEW

**Comment by Martin2112** [comment]
This week we have a special on merge conflicts ... I suppose it's got to be done!

File: cpp/fetcher/fetcher.cc [hunk 1]
```diff
@@ -146,7 +146,7 @@ void FetchState::WalkEntries() {
 
   int64_t index(start_);
   int num_fetch(0);
-  for (Range* current = entries_.get(); current;
+  for (Range *current = entries_.get(); current;
```
Comment (Martin2112) [inline cpp/fetcher/fetcher.cc]: I thought our style was Range* ?

File: cpp/log/cert.cc [hunk 2]
```diff
@@ -583,8 +583,8 @@ util::Status Cert::OctetStringExtensionData(int extension_nid,
   // Callers don't care whether extension is missing or invalid as they
   // usually call this method after confirming it to be present.
   const StatusOr<void*> ext_struct = ExtensionStructure(extension_nid);
-  if (!ext_struct.ok()
-      && ext_struct.status().CanonicalCode() == Code::NOT_FOUND) {
+  if (!ext_struct.ok() &&
```
Comment (Martin2112) [inline cpp/log/cert.cc]: Is that right? I thought we broke on operators, not after.

File: cpp/fetcher/fetcher.cc (hunk 1 above)
Comment (AlCutter) [inline cpp/fetcher/fetcher.cc]: It is :)
Weirdly, if we move `Range* current` out of the  `for` loop then it works as expected.
Bug in clang-format maybe?

File: cpp/log/cert.cc (hunk 2 above)
Comment (AlCutter) [inline cpp/log/cert.cc]: Don't think we had an opinion on this...
It's always been:
1) Google C++ Style Guide (which, AFAIR doesn't talk about exactly where to break)
2) then tie-break by falling back to what clang-format does.

File: cpp/client/ssl_client.cc [hunk 3]
```diff
@@ -189,7 +189,8 @@ int SSLClient::VerifyCallback(X509_STORE_CTX* ctx, void* arg) {
                 cert_trans::NID_ctSignedCertificateTimestampList)
           : util::Status::UNKNOWN;
 
-  // First check for embedded proof, if not present then look for the proof in a
+  // First check for embedded proof, if not present then look for the proof in
+  // a
```
Comment (pphaneuf) [inline cpp/client/ssl_client.cc]: Some awful here.

File: cpp/fetcher/remote_peer.cc [hunk 4]
```diff
@@ -83,7 +83,8 @@ void RemotePeer::Impl::DoneGetSTH(
     const LogVerifier::LogVerifyResult result(verifier_->VerifySignedTreeHead(
         *new_sth, 0, (time(NULL) + 10) * kNumMillisPerSecond));
 
-    // TODO(alcutter): We should probably log these invalid STHs somewhere a bit
+    // TODO(alcutter): We should probably log these invalid STHs somewhere a
+    // bit
```
Comment (pphaneuf) [inline cpp/fetcher/remote_peer.cc]: More awful.

File: cpp/log/cert.h [hunk 5]
```diff
@@ -164,9 +164,11 @@ class Cert {
   // you're doing.
   //
   // Returns OK if the extension data could be fetched and decoded.
-  // Returns NOT_FOUND if the extension is not present, or is present but is not
+  // Returns NOT_FOUND if the extension is not present, or is present but is
+  // not
```
Comment (pphaneuf) [inline cpp/log/cert.h]: Fix all the awful in this file.

File: cpp/log/ct_extensions.cc [hunk 6]
```diff
@@ -129,10 +129,10 @@ static X509V3_EXT_METHOD ct_poison_method = {
...
 // DNS id in the cert. See RFC section 3.2.2.
 static X509V3_EXT_METHOD ct_redaction_count_method = {
-    0,                         // ext_nid, NID, will be created by OBJ_create()
-    0,                         // flags
-    ASN1_ITEM_ref(REDACTED_LABEL_COUNT),
-    0, 0, 0, 0,                // ignored since the above is set
+    0,  // ext_nid, NID, will be created by OBJ_create()
+    0,  // flags
+    ASN1_ITEM_ref(REDACTED_LABEL_COUNT), 0, 0, 0,
+    0,  // ignored since the above is set
```
Comment (pphaneuf) [inline cpp/log/ct_extensions.cc]: This might be slightly wrong now?

File: cpp/log/ct_extensions.h [hunk 7]
```diff
@@ -10,12 +10,12 @@ void LoadCtExtensions();
 
 // Defines structure for redacted label count to hold a stack of integers
 typedef struct RedactedLabelCount_st {
-  STACK_OF(ASN1_INTEGER)* redacted_labels;
+  STACK_OF(ASN1_INTEGER) * redacted_labels;
```
Comment (pphaneuf) [inline cpp/log/ct_extensions.h]: Eeek! :smiley:

File: cpp/log/cert.cc (hunk 2 above)
Comment (Martin2112) [inline cpp/log/cert.cc]: OK. AFAIK the Google Java style guide does specify break before operator so it's what I'm used to seeing.

File: cpp/log/ct_extensions.cc (hunk 6 above)
Comment (Martin2112) [inline cpp/log/ct_extensions.cc]: Yes, those 4 values related to that comment.

File: cpp/log/ct_extensions.h (hunk 7 above)
Comment (AlCutter) [inline cpp/log/ct_extensions.h]: I guess we should look at replacing a lot of this with the new scoped ssl pointers and, if we're lucky, avoid this mess that way :)

I'll leave this as-is for now though...

File: cpp/log/ct_extensions.cc (hunk 6 above)
Comment (AlCutter) [inline cpp/log/ct_extensions.cc]: I've reverted the formatting on those blocks and slapped clang-format's wrist.

File: cpp/log/etcd_consistent_store-inl.h [hunk 8]
```diff
@@ -813,8 +812,8 @@ util::StatusOr<int64_t> EtcdConsistentStore<Logged>::CleanupOldEntries() {
     LOG(INFO) << "No current serving_sth, nothing to do.";
     return 0;
   }
-  const int64_t clean_up_to_sequence_number(
-      serving_sth_->Entry().tree_size() - 1);
+  const int64_t clean_up_to_sequence_number(serving_sth_->Entry().tree_size() -
```
Comment (Martin2112) [inline cpp/log/etcd_consistent_store-inl.h]: That looks weird?

File: cpp/log/cert.h (hunk 5 above)
Comment (AlCutter) [inline cpp/log/cert.h]: less awful'd

File: cpp/log/tree_signer-inl.h [hunk 9]
```diff
@@ -115,10 +115,10 @@ util::Status TreeSigner<Logged>::SequenceNewEntries() {
   for (const auto& m : mapping.Entry().mapping()) {
     // Go home clang-format, you're drunk.
     CHECK(
-        sequenced_hashes.insert(
-                             std::make_pair(m.entry_hash(),
-                                            std::make_pair(m.sequence_number(),
-                                                           false))).second);
+        sequenced_hashes
```
Comment (Martin2112) [inline cpp/log/tree_signer-inl.h]: I think that's been made less readable by the wonders of automation.

File: cpp/util/etcd.cc [hunk 10]
```diff
@@ -997,9 +1011,9 @@ void EtcdClient::MaybeLogEtcdVersion() {
   }
   logged_version_ = true;
 
-  const UrlFetcher::Request req(
-      URL("http://" + etcds_.front().first + ":" +
-              to_string(etcds_.front().second) + "/version"));
+  const UrlFetcher::Request req(URL("http://" + etcds_.front().first + ":" +
```
Comment (Martin2112) [inline cpp/util/etcd.cc]: It seems to have done the opposite with this than with some of the other changes.

**Comment by AlCutter** [comment]
Some things it's going to get "wrong" because it's not human, and while there is a vast array of knobs and levers you can twiddle with to try to get it less wrong, in practice all that happens is that you get the bit you're currently looking at "right" and something else goes wonky instead (but don't let that discourage you from having a pop at it if you're so minded ;)

Ultimately, though, it's better to have a tool which is the final arbiter on format, even if it's not 100% perfect.

File: cpp/util/etcd.cc [hunk 11]
```diff
@@ -41,9 +41,11 @@ using util::Task;
 
 DEFINE_int32(etcd_watch_error_retry_delay_seconds, 5,
              "delay between retrying etcd watch requests");
-DEFINE_bool(etcd_consistent, true, "Add consistent=true param to all requests. "
+DEFINE_bool(etcd_consistent, true,
+            "Add consistent=true param to all requests. "
```
Comment (pphaneuf) [inline cpp/util/etcd.cc]: Should put back the doc string in a single string, and let it re-split here?

File: cpp/util/etcd.cc [hunk 12]
```diff
@@ -41,9 +41,11 @@ using util::Task;
...
             "Do not turn this off unless you *know* what you're doing.");
-DEFINE_bool(etcd_quorum, true, "Add quorum=true param to all requests. "
+DEFINE_bool(etcd_quorum, true,
+            "Add quorum=true param to all requests. "
```
Comment (pphaneuf) [inline cpp/util/etcd.cc]: Same.

File: cpp/log/ct_extensions.h (hunk 7 above)
Comment (pphaneuf) [inline cpp/log/ct_extensions.h]: SGTM

**Comment by pphaneuf** [comment]
Done with the review.

**Comment by pphaneuf** [comment]
LGTM

**Comment by Martin2112** [comment]
LGTM, within reasonable doubt.

File: cpp/log/cert.cc (hunk 2 above)
Comment (benlaurie) [inline cpp/log/cert.cc]: I prefer it the old way. Apparently we can ask clang-format to do:

BreakBeforeBinaryOperators: BOS_NonAssignment

Comment (pphaneuf) [inline cpp/log/cert.cc]: I prefer the old way, but only by a small margin (with old one, operator is more obvious, with new one, nicer alignment), so I didn't might it too much. If there's a `clang-format` option to do it, sure, why not.

**Pull request merged**